src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -114,6 +116,13 @@ def validate(self, pdict=None):
                     msg = f"pileupName value {val} does not match dataset pattern"
                     self.logger.error(msg)
                     return False, msg
+            if key == 'customName' and val != '':
```

Comment:
I would suggest to use the `if` block above and reuse the same logic done for `pileupName`.

Reply:
done